File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/org/maplibre/android/maps/MaplibreTest.java
package org.maplibre.android.maps;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertSame;
import static org.junit.Assert.assertTrue;

import androidx.test.annotation.UiThreadTest;
import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner;

import org.maplibre.android.AppCenter;
import org.maplibre.android.MapLibre;
import org.maplibre.android.exceptions.MapLibreConfigurationException;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.ExpectedException;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4ClassRunner.class)
public class MapLibreTest extends AppCenter {

  private static final String API_KEY = "pk.0000000001";
  private static final String API_KEY_2 = "pk.0000000002";

  @Rule
  public ExpectedException expectedException = ExpectedException.none();

  private String realToken;

  @Before
  public void setup() {
    realToken = MapLibre.getApiKey();
  }

  @Test
  @UiThreadTest
  public void testConnected() {
    assertTrue(MapLibre.isConnected());

    // test manual connectivity
    MapLibre.setConnected(true);
    assertTrue(MapLibre.isConnected());
    MapLibre.setConnected(false);
    assertFalse(MapLibre.isConnected());

    // reset to Android connectivity
    MapLibre.setConnected(null);
    assertTrue(MapLibre.isConnected());
  }

  @Test
  @UiThreadTest
  public void setApiKey() {
    MapLibre.setApiKey(API_KEY);
    assertSame(API_KEY, MapLibre.getApiKey());
    MapLibre.setApiKey(API_KEY_2);
    assertSame(API_KEY_2, MapLibre.getApiKey());
  }

  @Test
  @UiThreadTest
  public void setNullApiKey() {
    expectedException.expect(MapLibreConfigurationException.class);
    expectedException.expectMessage(
      "A valid API key is required, currently provided key is: " + null
    );

    MapLibre.setApiKey(null);
  }

  @After
  public void tearDown() {
    MapLibre.setApiKey(realToken);
  }
}

File: platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/org/maplibre/android/maps/MapLibreTest.kt
package org.maplibre.android.maps

import androidx.test.annotation.UiThreadTest
import androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner
import org.junit.After
import org.junit.Assert.*
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.maplibre.android.AppCenter
import org.maplibre.android.MapLibre.*
import org.maplibre.android.exceptions.MapLibreConfigurationException

@RunWith(AndroidJUnit4ClassRunner::class)
class MapLibreTest : AppCenter() {
    private var realToken: String? = null
    @Before
    fun setup() {
        realToken = getApiKey()
    }

    @Test
    @UiThreadTest
    fun testConnected() {
        assertTrue(isConnected())

        // test manual connectivity
        setConnected(true)
        assertTrue(isConnected())
        setConnected(false)
        assertFalse(isConnected())

        // reset to Android connectivity
        setConnected(null)
        assertTrue(isConnected())
    }

    @Test
    @UiThreadTest
    fun setApiKey() {
        setApiKey(API_KEY)
        assertSame(API_KEY, getApiKey())
        setApiKey(API_KEY_2)
        assertSame(API_KEY_2, getApiKey())
    }

    @Test
    @UiThreadTest
    fun setNullApiKey() {
        assertThrows(MapLibreConfigurationException::class.java) { setApiKey(null) }
    }

    @After
    fun tearDown() {
        setApiKey(realToken)
    }

    companion object {
        private const val API_KEY = "pk.0000000001"
        private const val API_KEY_2 = "pk.0000000002"
    }
}